app/models/drug_stock.rb

```diff
@@ -7,9 +7,15 @@ class DrugStock < ApplicationRecord
   validates :received, numericality: true, allow_nil: true
   validates :for_end_of_month, presence: true
 
+  def self.latest_for_facilities(facilities, for_end_of_month)
+    DrugStock.select("DISTINCT ON (facility_id, protocol_drug_id) *")
+      .includes(:protocol_drug)
+      .where(facility: facilities.pluck(:id), for_end_of_month: for_end_of_month)
```

Comment:
`for_end_of_month: for_end_of_month` Looks like `for_end_of_month` is a full-fledged time stamp. Is this check _too_ precise? Is it prone to breaking if, say, we change the timezone that the dashboard operates in?

Reply:
Good point, hadn't thought about timezones, and haven't really looked at how we handle it. Will check and revert back. We are setting this value in the controller, so we're quite in control of it either way.

Reply:
We've changed the column type here to a date now with 134a75e731ed83a6b218b05aec352202a8c99876, so we shouldn't have a tz issue with this anymore. More over, the dates are chosen based on a dropdown during creation, and not based on relative time, so we needn't handle user specific timezones I think.